RHIDP-5422: TechDocs add-ons doc (#975)
redhat-developer/red-hat-developers-documentation-rhdh: modules/techdocs/proc-techdocs-addon-install-operator.adoc

// Module included in the following assemblies:
//
// * assemblies/assembly-techdocs-addons-installing.adoc

:_mod-docs-content-type: PROCEDURE
[id="proc-techdocs-addon-install-operator_{context}"]
== Installing and configuring a TechDocs add-on using the Operator

You can use the {product} Operator to install a TechDocs add-on on your {product} instance as a dynamic plugin.

.Prerequisites
* An administrator has installed the {product} Operator.
* You have installed the TechDocs plugin.

.Procedure

. From the Developer perspective in the {ocp-short} web console, click *ConfigMaps* > *Create ConfigMap*.
. From the *Create ConfigMap* page, select the *YAML view* option in the *Configure via* field.

> . In the the {product} ConfigMap, add the default `backstage-plugin-techdocs-module-addons-contrib` package configuration. For example:

Author: ```suggestion
. In the the {product} ConfigMap, add the default `backstage-plugin-techdocs-module-addons-contrib` package configuration. For example:
```

Is this the right place? Does the ConfigMap have a canonical file name we can reference?

Reviewer: You are describing here a procedure of creating new ConfigMap for plugins, so I would use something line 'in the newly created ConfigMap'.

+
[source,yaml,subs="+quotes,+attributes"]
----
kind: ConfigMap
apiVersion: v1
metadata:
  name: dynamic-plugins-rhdh
data:
  dynamic-plugins.yaml: |
    includes:
      - dynamic-plugins.default.yaml
    plugins:
      - package: ./dynamic-plugins/dist/backstage-plugin-techdocs-module-addons-contrib
        disabled: false
        pluginConfig:
          dynamicPlugins:
            frontend:
              backstage.plugin-techdocs-module-addons-contrib:
                techdocsAddons:
                  - importName: ReportIssue
----
+
[NOTE]
====
Preinstalled add-ons, such as `ReportIssue`, are included in the default package configuration. Additional add-ons are installed by manually adding them to the configuration file.
====

> . In the `techdocsAddons` section of the ConfigMap, add `importName: _<external_techdocs_add-on>` for each external TechDocs add-on that you want to install. For example:

Reviewer: ```suggestion
. In the `techdocsAddons` section of the ConfigMap, add `importName: _<external_techdocs_add-on>_` for each external TechDocs add-on that you want to use from the specified package above. For example:
```

If the package is specified, it will be installed first, then we import only specified add-ons from it.

Reviewer: Maybe configuring? The `contrib` package is preinstalled and we are overriding its default configuration to configure more add-ons from it.

Author: Maybe _adding_? I asked PM to share their opinions also.

Author: > Should we also update [preinstalled dynamic plugins ](https://redhat-developer.github.io/red-hat-developers-documentation-rhdh/release-1.5/plugin-rhdh/#con-preinstalled-dynamic-plugins)with @backstage/plugin-techdocs-module-addons-contrib

@Gerry-Forde should we create a Jira ticket for this?

> . In the `techdocsAddons` section of the ConfigMap, add `importName: _<external_techdocs_add-on>` for each external TechDocs add-on that you want to add from the specified plugin package. For example:

Reviewer: ```suggestion
. In the `techdocsAddons` section of the ConfigMap, add `importName: _<external_techdocs_add-on>_` for each external TechDocs add-on that you want to add from the specified plugin package. For example:
```
